include/deal.II/matrix_free/shape_info.h

```diff
@@ -285,6 +285,16 @@ namespace internal
        * end points of the unit cell.
        */
       bool nodal_at_cell_boundaries;
+
+      /**
+       * TODO
+       */
+      AlignedVector<Number> shape_values_face;
+
+      /**
+       * TODO
+       */
+      AlignedVector<Number> shape_gradients_face;
```

Comment:
I think we will need here also an array with pointers of into `shape_gradients_face`, which gives the right start point depending on face_no and orientation. This will be important once we have wedges and pyramids, cells with different number of quadrature points on each face.